### PR TITLE
:sparkles: return optimal text and line arguments

### PR DESCRIPTION
**Pull request by ThomasParistech**
It's sometimes useful to be able to store the optimal text boxes positions.
For instance, I use it to pre-compute optimal label positions once and then reuse-cache to draw at the same positions with different colors (via a GUI and a color-picker)

It won't break previous usage since nothing was returned.

### CODE REVIEW

File: src/textalloc/__init__.py [hunk 1]
```diff
@@ -75,6 +77,10 @@ def allocate(
         avoid_label_lines_overlap (bool, optional): If True, avoids overlap with lines drawn between text labels and locations. Defaults to False.
         plot_kwargs (dict, optional): kwargs for the plt.plot of the lines if draw_lines is True.
         **kwargs (): kwargs for the plt.text() call.
+
+    Returns:
+        result_text_xy (List[Optional[Tuple[float, float]]]): List of optimal (x,y) arguments to use when calling plt.text to draw the label.
+        result_line (List[Union[Optional[Tuple[float, float], Tuple[float, float]]]]): List of optimal (x,y) arguments to use when calling plt.plot to draw the line.
```
Comment (ckjellson) [inline src/textalloc/__init__.py:83]: Change to:
    Returns:
        result_text_xy (List[Optional[Tuple[float, float]]]): List of resulting (x,y) positions for text labels used in the plt.text call.
        result_lines (List[Union[Optional[Tuple[float, float], Tuple[float, float]]]]): List of resulting (x,y) pairs used in the plt.plot call for drawing lines.

**Review by ckjellson — CHANGES_REQUESTED** [review]
Very nice, thank you! It hurts my eyes a bit as a math-guy to see the word "optimal" here since the method will never be optimal 😅, I suggested some changes in the docs, and it would also be great if you could add something about it in the README after the parameters section, along the lines of:
"The allocate call returns a tuple containing the resulting positions used to plot the text labels and the connecting label lines"

Fix this and maybe change the commit-message as well to "return resulting text and line positions" and I will merge and update the pypi package!

**Comment by ThomasParistech** [comment]
I've applied your suggestion

You're right for the optimal adjective :smile:

**Comment by ckjellson** [comment]
Merged and updated pypi package

**Pull request merged**